src/core/crypto_tc.c

```diff
@@ -1974,7 +1974,7 @@ int32_t Crypto_TC_ProcessSecurity_Cam(uint8_t *ingest, int *len_ingest, TC_t *tc
         else  
         {
             // Some Magic here to log that an inappropriate SA was attempted to be used for EP
-            status = CRYPTO_LIB_ERR_SPI_INDEX_OOB;  // TODO:  Do we want a different error code for this?
+            status = CRYPTO_LIB_ERR_SDLS_EP_WRONG_SPI;  // TODO:  Do we want a different error code for this?
```

Comment:
Probably a good call on changing this error code to something more specific.  Should we remove that comment now?

Reply:
Good catch

Reply:
@rjbrown2 There is some weirdness here where we log an error status on 1978, then reset the final status to success. Could you ping me on that if you know/understand why it looks like that?

Reply:
That is done that way so that we log that we attempted to process a PDU, but did not because it wasn't the appropriate SA.  We log the error so that users would know what happened (without giving away too much other information), and then return successfully.

Reply:
We didn't set and return any failures here, as that would have required a lot more manipulation, and Donnie and I decided that this would be the best method for now without a big discussion...unless you disagree...then hack away.

Reply:
Makes sense. Sounds like by doing this and bubbling the error up it causes a lot of issues. I'll open a separate issue to track this.